src/microdot/microdot.py

```diff
@@ -774,7 +774,10 @@ def send_file(cls, filename, status_code=200, content_type=None,
         first.
         """
         if content_type is None:
-            ext = filename.split('.')[-1]
+            if filename.endswith('.gz'):
```

Comment:
```suggestion
            if compressed and filename.endswith('.gz'):
```

I think it is best to rely on the `compression` argument here.